{"rendered": {"description": {"raw": "While making a patch release of ign-gazebo1, I saw an error message at the end of the console output from `release.py`:\r\n\r\n~~~\r\n...\r\n- Linux: http://build.osrfoundation.org/job/ign-gazebo-debbuilder/buildWithParameters?...\r\nTraceback (most recent call last):\r\n  File \"/home/developer/release-tools/release.py\", line 535, in <module>\r\n    go(sys.argv)\r\n  File \"/home/developer/release-tools/release.py\", line 502, in go\r\n    for d in distros:\r\nTypeError: 'NoneType' object is not iterable\r\n~~~\r\n\r\nI believe this is because [ign-gazebo-release](https://bitbucket.org/ignitionrobotics/ign-gazebo-release/src) doesn't have debian/sid or debian/stretch metadata, only Ubuntu. This patch should suppress the complaint if a repo doesn't have any debian distro metadata.", "markup": "markdown", "html": "<p>While making a patch release of ign-gazebo1, I saw an error message at the end of the console output from <code>release.py</code>:</p>\n<div class=\"codehilite\"><pre><span></span>...\n- Linux: http://build.osrfoundation.org/job/ign-gazebo-debbuilder/buildWithParameters?...\nTraceback (most recent call last):\n  File &quot;/home/developer/release-tools/release.py&quot;, line 535, in &lt;module&gt;\n    go(sys.argv)\n  File &quot;/home/developer/release-tools/release.py&quot;, line 502, in go\n    for d in distros:\nTypeError: &#39;NoneType&#39; object is not iterable\n</pre></div>\n\n\n<p>I believe this is because <a data-is-external-link=\"true\" href=\"https://bitbucket.org/ignitionrobotics/ign-gazebo-release/src\" rel=\"nofollow\">ign-gazebo-release</a> doesn't have debian/sid or debian/stretch metadata, only Ubuntu. This patch should suppress the complaint if a repo doesn't have any debian distro metadata.</p>", "type": "rendered"}, "title": {"raw": "release.py: disable error message if no debian distro metadata is present", "markup": "markdown", "html": "<p>release.py: disable error message if no debian distro metadata is present</p>", "type": "rendered"}}, "type": "pullrequest", "description": "While making a patch release of ign-gazebo1, I saw an error message at the end of the console output from `release.py`:\r\n\r\n~~~\r\n...\r\n- Linux: http://build.osrfoundation.org/job/ign-gazebo-debbuilder/buildWithParameters?...\r\nTraceback (most recent call last):\r\n  File \"/home/developer/release-tools/release.py\", line 535, in <module>\r\n    go(sys.argv)\r\n  File \"/home/developer/release-tools/release.py\", line 502, in go\r\n    for d in distros:\r\nTypeError: 'NoneType' object is not iterable\r\n~~~\r\n\r\nI believe this is because [ign-gazebo-release](https://bitbucket.org/ignitionrobotics/ign-gazebo-release/src) doesn't have debian/sid or debian/stretch metadata, only Ubuntu. This patch should suppress the complaint if a repo doesn't have any debian distro metadata.", "links": {"decline": {"href": "https://api.bitbucket.org/2.0/repositories/osrf/release-tools/pullrequests/777/decline"}, "diffstat": {"href": "https://api.bitbucket.org/2.0/repositories/osrf/release-tools/diffstat/osrf/release-tools:1b98ded32de7%0De9992077d289?from_pullrequest_id=777"}, "commits": {"href": "data/repositories/osrf/release-tools/pullrequests/777/commits.json"}, "self": {"href": "data/repositories/osrf/release-tools/pullrequests/777.json"}, "comments": {"href": "data/repositories/osrf/release-tools/pullrequests/777/comments_page=1.json"}, "merge": {"href": "https://api.bitbucket.org/2.0/repositories/osrf/release-tools/pullrequests/777/merge"}, "html": {"href": "#!/osrf/release-tools/pull-requests/777"}, "activity": {"href": "data/repositories/osrf/release-tools/pullrequests/777/activity.json"}, "diff": {"href": "https://api.bitbucket.org/2.0/repositories/osrf/release-tools/diff/osrf/release-tools:1b98ded32de7%0De9992077d289?from_pullrequest_id=777"}, "approve": {"href": "https://api.bitbucket.org/2.0/repositories/osrf/release-tools/pullrequests/777/approve"}, "statuses": {"href": "data/repositories/osrf/release-tools/pullrequests/777/statuses_page=1.json"}}, "title": "release.py: disable error message if no debian distro metadata is present", "close_source_branch": true, "reviewers": [{"display_name": "Jose Luis Rivero", "uuid": "{d12309b2-f745-42ee-b119-aec4fcdf81fe}", "links": {"self": {"href": "https://api.bitbucket.org/2.0/users/%7Bd12309b2-f745-42ee-b119-aec4fcdf81fe%7D"}, "html": {"href": "https://bitbucket.org/%7Bd12309b2-f745-42ee-b119-aec4fcdf81fe%7D/"}, "avatar": {"href": "data/secure.gravatar.com/avatar/109284c8b83411dbc7492138f6167e9ed=httpsavatar-management--avatars.us-west-2.prod.public.atl-paas.netinitialsJR-5.png"}}, "nickname": "Jose Luis Rivero", "type": "user", "account_id": "557058:155a32e2-420c-4d50-98e0-0e722f63f906"}], "id": 777, "destination": {"commit": {"hash": "e9992077d289", "type": "commit", "links": {"self": {"href": "data/repositories/osrf/release-tools/commit/e9992077d289.json"}, "html": {"href": "#!/osrf/release-tools/commits/e9992077d289"}}}, "repository": {"links": {"self": {"href": "data/repositories/osrf/release-tools.json"}, "html": {"href": "#!/osrf/release-tools"}, "avatar": {"href": "data/bytebucket.org/ravatar/{41811d3d-19d0-428e-a134-976595e72508}ts=python"}}, "type": "repository", "name": "release-tools", "full_name": "osrf/release-tools", "uuid": "{41811d3d-19d0-428e-a134-976595e72508}"}, "branch": {"name": "default"}}, "created_on": "2019-03-12T22:52:15.793684+00:00", "summary": {"raw": "While making a patch release of ign-gazebo1, I saw an error message at the end of the console output from `release.py`:\r\n\r\n~~~\r\n...\r\n- Linux: http://build.osrfoundation.org/job/ign-gazebo-debbuilder/buildWithParameters?...\r\nTraceback (most recent call last):\r\n  File \"/home/developer/release-tools/release.py\", line 535, in <module>\r\n    go(sys.argv)\r\n  File \"/home/developer/release-tools/release.py\", line 502, in go\r\n    for d in distros:\r\nTypeError: 'NoneType' object is not iterable\r\n~~~\r\n\r\nI believe this is because [ign-gazebo-release](https://bitbucket.org/ignitionrobotics/ign-gazebo-release/src) doesn't have debian/sid or debian/stretch metadata, only Ubuntu. This patch should suppress the complaint if a repo doesn't have any debian distro metadata.", "markup": "markdown", "html": "<p>While making a patch release of ign-gazebo1, I saw an error message at the end of the console output from <code>release.py</code>:</p>\n<div class=\"codehilite\"><pre><span></span>...\n- Linux: http://build.osrfoundation.org/job/ign-gazebo-debbuilder/buildWithParameters?...\nTraceback (most recent call last):\n  File &quot;/home/developer/release-tools/release.py&quot;, line 535, in &lt;module&gt;\n    go(sys.argv)\n  File &quot;/home/developer/release-tools/release.py&quot;, line 502, in go\n    for d in distros:\nTypeError: &#39;NoneType&#39; object is not iterable\n</pre></div>\n\n\n<p>I believe this is because <a data-is-external-link=\"true\" href=\"https://bitbucket.org/ignitionrobotics/ign-gazebo-release/src\" rel=\"nofollow\">ign-gazebo-release</a> doesn't have debian/sid or debian/stretch metadata, only Ubuntu. This patch should suppress the complaint if a repo doesn't have any debian distro metadata.</p>", "type": "rendered"}, "source": {"commit": {"hash": "af0455a81b86", "type": "commit", "links": {"self": {"href": "data/repositories/osrf/release-tools/commit/af0455a81b86.json"}, "html": {"href": "#!/osrf/release-tools/commits/af0455a81b86"}}}, "repository": {"links": {"self": {"href": "data/repositories/osrf/release-tools.json"}, "html": {"href": "#!/osrf/release-tools"}, "avatar": {"href": "data/bytebucket.org/ravatar/{41811d3d-19d0-428e-a134-976595e72508}ts=python"}}, "type": "repository", "name": "release-tools", "full_name": "osrf/release-tools", "uuid": "{41811d3d-19d0-428e-a134-976595e72508}"}, "branch": {"name": "no_debian_error"}}, "comment_count": 1, "state": "MERGED", "task_count": 0, "participants": [{"role": "REVIEWER", "participated_on": "2019-03-13T16:00:37.828871+00:00", "type": "participant", "approved": true, "user": {"display_name": "Jose Luis Rivero", "uuid": "{d12309b2-f745-42ee-b119-aec4fcdf81fe}", "links": {"self": {"href": "https://api.bitbucket.org/2.0/users/%7Bd12309b2-f745-42ee-b119-aec4fcdf81fe%7D"}, "html": {"href": "https://bitbucket.org/%7Bd12309b2-f745-42ee-b119-aec4fcdf81fe%7D/"}, "avatar": {"href": "data/secure.gravatar.com/avatar/109284c8b83411dbc7492138f6167e9ed=httpsavatar-management--avatars.us-west-2.prod.public.atl-paas.netinitialsJR-5.png"}}, "nickname": "Jose Luis Rivero", "type": "user", "account_id": "557058:155a32e2-420c-4d50-98e0-0e722f63f906"}}], "reason": "", "updated_on": "2019-03-13T16:00:49.199792+00:00", "author": {"display_name": "Steve Peters", "uuid": "{2ccfed09-18b8-4921-8d58-15ef01092802}", "links": {"self": {"href": "https://api.bitbucket.org/2.0/users/%7B2ccfed09-18b8-4921-8d58-15ef01092802%7D"}, "html": {"href": "https://bitbucket.org/%7B2ccfed09-18b8-4921-8d58-15ef01092802%7D/"}, "avatar": {"href": "data/secure.gravatar.com/avatar/1fb4816dad9e68337d3096f750951f6cd=httpsavatar-management--avatars.us-west-2.prod.public.atl-paas.netinitialsSP-1.png"}}, "nickname": "Steven Peters", "type": "user", "account_id": "557058:5de38267-b118-494c-aa76-4fab35448816"}, "merge_commit": {"hash": "1b98ded32de7", "type": "commit", "links": {"self": {"href": "data/repositories/osrf/release-tools/commit/1b98ded32de7.json"}, "html": {"href": "#!/osrf/release-tools/commits/1b98ded32de7"}}}, "closed_by": {"display_name": "Jose Luis Rivero", "uuid": "{d12309b2-f745-42ee-b119-aec4fcdf81fe}", "links": {"self": {"href": "https://api.bitbucket.org/2.0/users/%7Bd12309b2-f745-42ee-b119-aec4fcdf81fe%7D"}, "html": {"href": "https://bitbucket.org/%7Bd12309b2-f745-42ee-b119-aec4fcdf81fe%7D/"}, "avatar": {"href": "data/secure.gravatar.com/avatar/109284c8b83411dbc7492138f6167e9ed=httpsavatar-management--avatars.us-west-2.prod.public.atl-paas.netinitialsJR-5.png"}}, "nickname": "Jose Luis Rivero", "type": "user", "account_id": "557058:155a32e2-420c-4d50-98e0-0e722f63f906"}}